scripts/apps/authoring/authoring/directives/ItemCarouselDirective.ts

```diff
@@ -77,12 +86,13 @@ export function ItemCarouselDirective(notify) {
              * otherwise carousel height is messed up
              */
             scope.$watch('items', (items: Array<any>) => {
-                // Don't execute if there are no items or their length is same as before
-                if (items == null || previousItems && getItemsCount(items) === getItemsCount(previousItems)) {
+                // Don't execute if there are no items or their length is same as before and their order is unchanged
+                if (items == null || previousItems && getItemsCount(items) === getItemsCount(previousItems)
```

Comment:
wouldn't [$watchCollection](https://docs.angularjs.org/api/ng/type/$rootScope.Scope#$watchCollection) do the same thing?

Reply:
Do you mean replacing $watch with $watchCollection?

Reply:
exactly